### PR TITLE
Change .merlin generation to not use PKG

### PR DESCRIPTION
**Pull request by rgrinberg**
We know the build directory of exteranl packages, hence we should simply use it.

This is extracted from #498 . It's a big enough change to give it its own commit here. I've tested this myself on a couple of projects and it seems to work the same.

### CODE REVIEW

**Comment by diml** [comment]
Does the `jump to definition` still works? I would have thought we needed a `S` directive for that.

File: src/merlin.ml [hunk 1]
```diff
@@ -48,18 +48,16 @@ let dot_merlin sctx ~dir ~scope ({ requires; flags; _ } as t) =
       requires &&& flags
       >>^ (fun (libs, flags) ->
         let ppx_flags = ppx_flags sctx ~dir ~scope ~src_dir:remaindir t in
-        let internals, externals =
-          List.fold_left libs ~init:([], []) ~f:(fun (internals, externals) ->
+        let libs =
+          List.fold_left libs ~init:[] ~f:(fun acc ->
+            let nice_path = Path.reach ~from:remaindir in
```
Comment (diml) [inline src/merlin.ml]: I'm not sure `nice` gives the right idea. The idea is more to serialize an internal representation of a path. Maybe `serialize_path`?

**Comment by rgrinberg** [comment]
We do have an `S` directive for internal libs. Should we do it for external libs as well? I suppose we do install all source files now.

**Comment by diml** [comment]
I would think we need to but I'm not 100% sure. I would think that `jump to definition` is affected by this, so I think it's a good test

**Comment by rgrinberg** [comment]
> I would think we need to but I'm not 100% sure. I would think that jump to definition is affected by this, so I think it's a good test

I added it by I couldn't see an improvement in my tests. merlin is able to find module definition of external libraries, but not the definitions of individual functions. This is consistent with any of the following options: `PKG`, just `B`, `S` & `B`. cc @trefis

**Comment by trefis** [comment]
Everything added as `PKG` at the moment currently goes to both `B` and `S` so I think you want to do the same.

As for the rest, I don't know what your test cases are, feel free to open issues on merlin's bugtracker if you think things aren't working as they should be :)

**Comment by rgrinberg** [comment]
OK, so it's settled. We'll use both `S` and `B`. Let me squash my commits.

**Pull request merged**